pkg/apis/apps/validation/validation_test.go

```diff
@@ -36,6 +36,7 @@ import (
 
 func TestValidateStatefulSet(t *testing.T) {
 	validLabels := map[string]string{"a": "b"}
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.MaxUnavailableStatefulSet, true)()
```

Comment:
There is no dependency on the feature gate. Remove this line.

Reply:
But the validation function itself doesn't use this feature gate?

It's much more important to test things with the feature gate off, since that's how most clusters will run.